packages/chat-stateful-client/src/ResourceDownloadQueue.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { ChatContext } from './ChatContext';
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { ChatError } from './ChatClientState';
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { ChatMessageWithStatus } from './types/ChatMessageWithStatus';
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { CommunicationTokenCredential } from '@azure/communication-common';
+
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+/**
+ * @private
+ */
+export class ResourceDownloadQueue {
+  private _messageQueue: ChatMessageWithStatus[] = [];
+  private _context: ChatContext;
+  private _isRequesting = false;
+  private _credential: CommunicationTokenCredential;
+
+  constructor(context: ChatContext, credential: CommunicationTokenCredential) {
+    this._context = context;
+    this._credential = credential;
+  }
+
+  public isRequesting(): boolean {
```

Comment:
Why do we need this exposed publicly?
Could the checks for the state of the queue and whether the message is already added be encapsulated within the class and simplify the ergonomics from the perspective of the caller?

Reply:
Yep agree. It makes sense to just have this check internally when we start the queue in case caller calls it mistakenly multiple times